File: Domain/AppStatusList.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain
{
    public class AppStatusList
    {
      public AppStatusList()
      {
        this.ActionList = new HashSet<AppAction>();
      }

      [Key]
      [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
      public int Id { get; set; }
      public string Title { get; set; }
      public int Order { get; set; }	
      public int TableId { get; set; }
      public virtual ICollection<AppAction> ActionList { get; set; }      
    }
}

File: client-app/src/features/AppStatusList/AppStatusList.ts
export interface IAppStatusList {
	Id: number
	Title: string
	Order: number
	TableId: number
}

export class AppStatusList implements IAppStatusList {
	Id: number = 0;
	Title: string = '';
	Order: number = 0;
	TableId: number = 0;
  
  constructor(init?: IAppStatusList) {
    Object.assign(this, init);
  }
}

